keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -61,26 +62,58 @@ class ClusterHandler(
     }
 
   override fun upsert(resource: Resource<Cluster>, diff: DiffNode?) {
-    val taskRef = runBlocking {
+    runBlocking {
       val spec = resource.spec
       val job = when {
         diff.isCapacityOnly() -> spec.resizeServerGroupJob()
-        else -> spec.createServerGroupJob()
+        else -> spec.createServerGroupJob().let { job ->
```

Comment:
General style comment - I dislike chaining big functions with the `.let{}` / `when` etc. I prefer something like 
```
else -> {
  val job = spec.createServerGroupJob()
  when {
    job["amiName"] != null -> job
   resource.metadata.data["packageRef"] != null ->  {
     //a function for what you're doing with the if/else
   }
  }
}
```
I find the very chained style pretty hard to read and reason about when you're looking through someone else's code.

Reply:
I've reworked this to pull more stuff out into functions but I would really welcome any more `this is too hard to read/reason about` feedback on where I landed. This kotlin thing is still pretty shiny for me so I am not entirely sure which features I like or dislike for terseness vs readability